cryptography/src/bls12381/primitives/ops.rs

```diff
@@ -130,7 +144,10 @@ pub fn verify_aggregate(
         messages
             .par_iter()
             .map(|msg| {
-                let payload = union_unique(namespace, msg);
+                let payload = match namespace {
```

Comment:
To avoid unnecessary conversion of `msg.to_vec()`, I suggest we do:
```
let mut hm = group::Signature::zero();
match namespace {
  Some(namespace) => { let payload = union_unique(namespace, msg); hm.map(&payload); },
  None => hm.map(&msg);
}
hm
```

I guess this is slightly more core reuse (in that we call `.map()` in two places) but think its worth it for the memory efficiency.

Reply:
There may be other places where you did it this way that I suggest we modify.